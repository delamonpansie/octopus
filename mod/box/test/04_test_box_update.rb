#!/usr/bin/ruby

$: << File.dirname($0) + '/lib'
require 'run_env'

RunEnv.connect_eval do
  ping
  insert [1, 2, "abc", "def"]

  update_fields 1, [1, :or, 0xff]
  select 1

  update_fields 1, [1, :set, 127]
  select 1

  update_fields 1, [2, :splice, 2, 4, "aaaa"]
  select 1

  update_fields 1, [2, :splice, 2, 4, ""]
  select 1

  update_fields 1, [1, :insert, 11]
  select 1

  update_fields 1, [1, :delete, ""]
  select 1

  update_fields 1, [1, :delete, ""], [1, :delete, ""], [1, :delete, ""]
  select 1

  log_try { update_fields 1, [1, :delete, ""] }
  select 1

  insert [1]
  update_fields 1, [1, :insert, "aa"]
  select 1

  insert [10, "foo"]
  update_fields 10, [1, :splice, 0, 1, "b"]
  select 10

  insert [10, "foo"]
  update_fields 10, [1, :set, ""]
  update_fields 10, [1, :set, ""]
  select 10

  # test update of PK
  insert [2, "foo"]
  insert [3, "bar"]
  select 2, 3, 4
  update_fields 2, [0, :set, 4]
  select 2, 3, 4
  log_try { update_fields 4, [0, :set, 3] }
end
