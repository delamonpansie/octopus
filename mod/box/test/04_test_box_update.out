# box.ping()
:pong

# box.insert([1, 2, "abc", "def"])
1

# box.update_fields(1, [1, :or, 255])
1

# box.select(1)
[["\x01\x00\x00\x00", "\xFD\x00\x00\x00", "abc", "def"]]

# box.update_fields(1, [1, :set, 127])
1

# box.select(1)
[["\x01\x00\x00\x00", "\x7F\x00\x00\x00", "abc", "def"]]

# box.update_fields(1, [2, :splice, 2, 4, "aaaa"])
1

# box.select(1)
[["\x01\x00\x00\x00", "\x7F\x00\x00\x00", "abaaaa", "def"]]

# box.update_fields(1, [2, :splice, 2, 4, ""])
1

# box.select(1)
[["\x01\x00\x00\x00", "\x7F\x00\x00\x00", "ab", "def"]]

# box.update_fields(1, [1, :insert, 11])
1

# box.select(1)
[["\x01\x00\x00\x00", "\v\x00\x00\x00", "\x7F\x00\x00\x00", "ab", "def"]]

# box.update_fields(1, [1, :delete, ""])
1

# box.select(1)
[["\x01\x00\x00\x00", "\x7F\x00\x00\x00", "ab", "def"]]

# box.update_fields(1, [1, :delete, ""], [1, :delete, ""], [1, :delete, ""])
1

# box.select(1)
[["\x01\x00\x00\x00"]]

# box.update_fields(1, [1, :delete, ""])
Failed with: {code: 0x202, message: 'update of field beyond tuple cardinality'}
# box.select(1)
[["\x01\x00\x00\x00"]]

# box.insert([1])
2

# box.update_fields(1, [1, :insert, "aa"])
1

# box.select(1)
[["\x01\x00\x00\x00", "aa"]]

# box.insert([10, "foo"])
1

# box.update_fields(10, [1, :splice, 0, 1, "b"])
1

# box.select(10)
[["\n\x00\x00\x00", "boo"]]

# box.insert([10, "foo"])
2

# box.update_fields(10, [1, :set, ""])
1

# box.update_fields(10, [1, :set, ""])
1

# box.select(10)
[["\n\x00\x00\x00", ""]]

# box.insert([2, "foo"])
1

# box.insert([3, "bar"])
1

# box.select(2, 3, 4)
[["\x02\x00\x00\x00", "foo"], ["\x03\x00\x00\x00", "bar"]]

# box.update_fields(2, [0, :set, 4])
2

# box.select(2, 3, 4)
[["\x03\x00\x00\x00", "bar"], ["\x04\x00\x00\x00", "foo"]]

# box.update_fields(4, [0, :set, 3])
Failed with: {code: 0x3802, message: 'duplicate key value violates unique index 0:GenHash'}
